Create python sdk (#1)
eclipse-ankaios/ank-sdk-python: README.md

## Installation

### Install via pip

```sh
pip install ankaios-sdk
```

### Clone and Local Build

```sh
# Clone repository
git clone https://github.com/eclipse-ankaios/ank-sdk-python.git
cd ank-sdk-python

# Install in editable mode
pip install -e .

# If you plan on contributing or running tests locally
pip install -e .[dev]
```

## Usage

After installation, you can use the Ankaios SDK to configure and run workloads and request
the state of the Ankaios system and the connected agents.

Example:
```python
from ankaios_sdk import Workload, Ankaios, WorkloadStateEnum, WorkloadSubStateEnum

# Connect to control interface
with Ankaios() as ankaios:
  # Create a new workload
  workload = Workload.builder() \
    .workload_name("dynamic_nginx") \
    .agent_name("agent_A") \
    .runtime("podman") \
    .restart_policy("NEVER") \
    .runtime_config("image: docker.io/library/nginx\ncommandOptions: [\"-p\", \"8080:80\"]") \
    .build()

  # Run the workload
  ret = ankaios.run_workload(workload)

  # Check if the workload is scheduled and get the WorkloadInstanceName
  if ret is not None:
    workload_instance_name = ret["added_workloads"][0]
  

>   # Wait until the workload raches the running state

Reviewer: ```suggestion
  # Wait until the workload reaches the running state
```

  ret = ankaios.wait_for_workload_to_reach_state(
    workload_instance_name,
    state=WorkloadStateEnum.RUNNING,
    timeout=5
    )
  if ret:
    print("Workload reached the RUNNING state.")

  # Request the workload state based on the workload instance name
  ret = ankaios.get_workload_state_for_instance_name(workload_instance_name)
  if ret is not None:

>     print(f"State: {ret.state}, substate: {ret.substate}, info: {ret.info}")

Author: Done.

```

### Clone and Local Build

```sh
# Clone repository
git clone https://github.com/eclipse-ankaios/ank-sdk-python.git
cd ank-sdk-python

# Install in editable mode
pip install -e .

# If you plan on contributing or running tests locally
pip install -e ".[dev]"
```

> **Note:**  
> Depending on your Linux distribution, it could be that you need to create and activate a [virtual environment](https://docs.python.org/3/library/venv.html) to run the pip commands.

## Usage

After installation, you can use the Ankaios SDK to configure and run workloads and request
the state of the Ankaios system and the connected agents.

Example:
```python
from ankaios_sdk import Workload, Ankaios, WorkloadStateEnum, WorkloadSubStateEnum

# Create a new Ankaios object.
# The connection to the control interface is automatically done at this step.
ankaios = Ankaios()

# Create a new workload
workload = Workload.builder() \
  .workload_name("dynamic_nginx") \
  .agent_name("agent_A") \
  .runtime("podman") \
  .restart_policy("NEVER") \
  .runtime_config("image: docker.io/library/nginx\ncommandOptions: [\"-p\", \"8080:80\"]") \
  .build()

# Run the workload
ret = ankaios.apply_workload(workload)

> # Check if the workload is scheduled and get the WorkloadInstanceName

Reviewer 2: ```suggestion
# Get the WorkloadInstanceName to check later if the workload is running.
```

Author: Done.

# Ankaios Python SDK

Eclipse Ankaios Python SDK provides a convenient python interface for interacting with the Ankaios platform.

# To do
- Finish the marked TODO methods from Ankaios.py, realted to configuration handling.

> - Add to pip wheel

Author: Changed the ReadMe, now only the config implementation remains 😄

<picture style="padding-bottom: 1em;">
  <source media="(prefers-color-scheme: dark)" srcset="https://github.com/eclipse-ankaios/ankaios/blob/main/logo/Ankaios__logo_for_dark_bgrd_clipped.png">
  <source media="(prefers-color-scheme: light)" srcset="https://github.com/eclipse-ankaios/ankaios/blob/main/logo/Ankaios__logo_for_light_bgrd_clipped.png">
  <img alt="Shows Ankaios logo" src="https://github.com/eclipse-ankaios/ankaios/blob/main/logo/Ankaios__logo_for_light_bgrd_clipped.png">
</picture>

# Ankaios Python SDK for Eclipse Ankaios

Eclipse Ankaios provides workload and container orchestration for automotive

> High Performance Computing Software (HPCs). While it can be used for various

Author: This was copied directly from Ankaios README.md:
https://github.com/eclipse-ankaios/ankaios/blob/76d5773e1d0edf64978fcc75cfeea089dcccddf9/README.md?plain=1#L10
If we change here, we should change in Ankaios as well I think.

> High Performance Computing Software (HPCs). While it can be used for various

Reviewer 2: There are also other changes that need to be synced with ankaios. We should open a PR and do all of then at once.

with Ankaios() as ankaios:
  # Create a new workload
  workload = Workload.builder() \
    .workload_name("dynamic_nginx") \
    .agent_name("agent_A") \
    .runtime("podman") \
    .restart_policy("NEVER") \
    .runtime_config("image: docker.io/library/nginx\ncommandOptions: [\"-p\", \"8080:80\"]") \
    .build()

  # Run the workload
  ankaios.run_workload(workload)

  # Request the state of the system, filtered with the current workload
  complete_state = ankaios.get_state(
    timeout=5,
    field_mask=["workloadStates.agent_A.dynamic_nginx"])

  # Get the workload states present in the complete_state
  workload_states_dict = complete_state.get_workload_states().get_as_dict()

  # Get the state of the desired workload
  dynamic_nginx_state = workload_states_dict["agent_A"]["dynamic_nginx"].values()[0]

  # Check state
  if dynamic_nginx_state.state == WorkloadStateEnum.RUNNING and
    dynamic_nginx_state.substate == WorkloadSubStateEnum.RUNNING_OK:
    print("Workload started running succesfully")
  elif dynamic_nginx_state.state == WorkloadStateEnum.FAILED:
    print("Workload failed with the following substate: {}".format(
      dynamic_nginx_state.substate.name

>     ))

Author: Redone it. It should look better now 😄
